### PR TITLE
Enrich .gitignore

### PR DESCRIPTION
**Pull request by aamederen**
This change provides nicer results for `git status` command. The following
items are added to .gitignore:
- Eclipse CDT project files
- Uncrustify backup files
- Output files of `perf`

### CODE REVIEW

**Comment by anarazel** [comment]
Personally I think the output of random tools is better ignored locally on systems; but on the other hand it doesn't cost much.

**Comment by jasonmp85** [comment]
Yeah, so if we're going to refactor the `gitignore` files, I'd prefer to make some guidelines and follow them. Either (a) separate files for each subdirectory, or (b) a global `gitignore` at the top. Adding a few tool rules doesn't seem worthwhile.

In the past I've followed (b) and I separate my `gitignore`s into [sections for each type of language and tool](https://github.com/citusdata/pg_shard/blob/develop/.gitignore). The language/tool sections are copy-pasted from [this repository](https://github.com/github/gitignore) and I keep a small "project-specific" section at the bottom, which is the user-editable part.

But these tools and sections are for languages and tools _unarguably_ used by the project (either it's written in them, or has dependencies on them, or its build process uses them).

For tools that _developers_ choose to use, I use a [global `gitignore`](https://help.github.com/articles/ignoring-files/#create-a-global-gitignore). Mine looks like this:

```gitignore
# ===========
# = Eclipse =
# ===========
*.pydevproject
.metadata
.gradle
bin/
tmp/
*.tmp
*.bak
*.swp
*~.nib
local.properties
.settings/
.loadpath

# Eclipse Core
.project

# External tool builders
.externalToolBuilders/

# Locally stored "Eclipse launch configurations"
*.launch

# CDT-specific
.cproject

# JDT-specific (Eclipse Java Development Tools)
.classpath

# Java annotation processor (APT)
.factorypath

# PDT-specific
.buildpath

# sbteclipse plugin
.target

# TeXlipse plugin
.texlipse

# STS (Spring Tool Suite)
.springBeans

# =======
# = OSX =
# =======
.DS_Store
.AppleDouble
.LSOverride

# Icon must end with two \r
Icon


# Thumbnails
._*

# Files that might appear in the root of a volume
.DocumentRevisions-V100
.fseventsd
.Spotlight-V100
.TemporaryItems
.Trashes
.VolumeIcon.icns

# Directories potentially created on remote AFP share
.AppleDB
.AppleDesktop
Network Trash Folder
Temporary Items
.apdisk

# ============
# = TextMate =
# ============
*.tmproj
*.tmproject
tmtags

# ================
# = My Overrides =
# ================

# Some projects might want this
!bin/
```
 As `citus_indent` doesn't actually produce any backup files in its default mode, and as Eclipse is a developer choice, I think those two are better served by a global `gitignore`.

And I assume `perf` is also something not specifically needed by the project (unless I missed some patch or something related to it). Can all of these changes just live in your global `gitignore`?

**Comment by jasonmp85** [comment]
tl;dr version: `git status` should always show no files changed for:

  * A fresh checkout
  * After first running `./configure`
  * After running `make all` or `make test`
  * After running `make clean`

But if a developer adds files because of tools they've chosen, our project can't reasonably predict all of them, so they're better handled by an external/global `gitignore` managed by that developer.

**Comment by aamederen** [comment]
Suggestions from @jasonmp85 and @anarazel are cool and I am closing this PR. Thanks.
